### PR TITLE
Add support to configure index

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

The sdk lacks the ability to configure an index.

## Solution

Added support to configure `podtype` and the `number of replicas` for an index.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Added unit and integrations tests.

### CODE REVIEW

File: src/test/java/io/pinecone/PineconeConnectionTest.java [hunk 1]
```diff
@@ -1,8 +1,5 @@
 package io.pinecone;
 
-import io.pinecone.PineconeClientConfig;
```
Comment (rohanshah18) [inline src/test/java/io/pinecone/PineconeConnectionTest.java:3]: Removed unnecessary imports.

File: src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java [hunk 2]
```diff
@@ -46,6 +47,18 @@ public void createAndDelete() throws IOException {
         List<String> indexList = pinecone.listIndexes();
         assert !indexList.isEmpty();
 
+        // Configure the index
```
Comment (rohanshah18) [inline src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java:50]: This is integration test where I'm configuring podtype and replicas which are further verified by describeIndex call.

File: src/test/java/io/pinecone/PineconeIndexOperationClientTest.java [hunk 3]
```diff
@@ -265,4 +265,31 @@ public void testListIndexes() throws IOException {
         assertNotNull(emptyIndexList);
         assertTrue(emptyIndexList.isEmpty());
     }
+
+    @Test
+    public void testConfigureIndex() throws IOException {
```
Comment (rohanshah18) [inline src/test/java/io/pinecone/PineconeIndexOperationClientTest.java:270]: Added unit test here

File: src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java (hunk 2 above)
Comment (jhamon) [reply src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java:50]: In tests like this, I like to assert what the state was before applying a change. Just to make sure my test is actually doing what I think. For example, [here](https://github.com/pinecone-io/pinecone-ts-client/blob/main/src/integration/control/configureIndex.test.ts#L89), in the typescript integration tests.

File: src/main/java/io/pinecone/PineconeIndexOperationClient.java [hunk 4]
```diff
@@ -96,9 +107,11 @@ private Request buildRequest(String method, String path, String acceptHeader, Re
         if (HTTP_METHOD_POST.equals(method)) {
             builder.post(requestBody);
             builder.addHeader(CONTENT_TYPE, CONTENT_TYPE_JSON);
-        } else if (HTTP_METHOD_DELETE.equals(method)) {
+        } else if(HTTP_METHOD_PATCH.equals(method)) {
+            builder.patch(requestBody);
+            builder.addHeader(CONTENT_TYPE, CONTENT_TYPE_JSON);
+        } else if (HTTP_METHOD_DELETE.equals(method))
             builder.delete();
-        }
```
[the remark this answers is not in the document]
Comment (rohanshah18) [reply src/main/java/io/pinecone/PineconeIndexOperationClient.java:101]: I actually skipped the brackets completely by mistake which is fine for java since there's only one line after the else-if but definitely not consistent, so let me add it back

**Review by austin-denoble — APPROVED** [review]
Nice work! Agreed with Jen, would be nice to add some error-case tests if it's not too much trouble.

File: src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java (hunk 2 above)
Comment (rohanshah18) [reply src/integration/java/io/pinecone/PineconeIndexOperationsClientIntegrationTest.java]: Added integration tests with edge cases!
